eng/Generation.psm1

```diff
@@ -50,6 +50,59 @@ function AutoRest-Reset()
     Invoke "$script:autoRestBinary --reset"
 }
 
+function Invoke-Cadl($baseOutput, $projectName, $sharedSource="", $fast="", $debug="")
+{
+    $baseOutput = Resolve-Path -Path $baseOutput
+    $baseOutput = $baseOutput -replace "\\", "/"
+    $outputPath = Join-Path $baseOutput "Generated"
+    $outputPath = $outputPath -replace "\\", "/"
+    #clean up
+    if (Test-Path $outputPath) {
+        Remove-Item $outputPath/* -Include *.* -Exclude *Configuration.json*
+    }
+    # emit cadl json
+    $repoRootPath = Join-Path $PSScriptRoot ".."
+    $repoRootPath = Resolve-Path -Path $repoRootPath
+    Push-Location $repoRootPath
+    # node node_modules\@cadl-lang\compiler\dist\core\cli.js compile --output-path $outputPath "$baseOutput\$projectName.cadl" --emit @azure-tools/cadl-csharp
+    $emitCommand = "node node_modules/@cadl-lang/compiler/dist/core/cli.js compile --output-path $outputPath $baseOutput/$projectName.cadl --emit @azure-tools/cadl-csharp"
+    Invoke $emitCommand
+    Pop-Location
+    
+    $dotnetArguments = $debug ? "--no-build --debug" : "--no-build" 
+    $command = "dotnet run --project $script:AutoRestPluginProject $dotnetArguments --standalone $outputPath"
+    Invoke $command
+    Invoke "dotnet build $baseOutput --verbosity quiet /nologo"
+}
+
+function Invoke-CadlSetup()
+{
+    # build emitter
+    $emitterPath = Join-Path $PSScriptRoot ".." "src" "CADL.Extension" "Emitter.Csharp"
+    $emitterPath = Resolve-Path -Path $emitterPath
+    Push-Location $emitterPath
+    npm install
+    npm run build
+    npm pack
+    Pop-Location
+
+    # install cadl and emitter
+    $repoRoot = Join-Path $PSScriptRoot ".."
+    $repoRoot = Resolve-Path $repoRoot
+    Push-Location $repoRoot
+    $packages = Get-ChildItem $repoRoot -Filter azure-tools-cadl-csharp-*.tgz -Recurse | Select-Object -ExpandProperty FullName | Resolve-Path -Relative
```

Comment:
What is azure-tools-cadl-csharp-*.tgz? I thought we should publish it?

Reply:
This is CI, we need to use the one local generated version, not the published one.

Reply:
I see, then where do you pack this *tgz?

Reply:
Is it possible that copy the built emitter to folder `node_modules` so that we don't need to pack it?

Reply:
Yes, copy is an workaround, but it is not suggested way to install an npm library. `npm link` or symbol link may be the solution. Now pack the library and install it is suitable way.

Reply:
I agree, especially `node_modules` is in gitignore.